File: java/src/test/java/problems101_200/WordBreak2Test.java
package problems101_200;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class WordBreak2Test extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "word-break-ii.json";
    }

    @Test
    public void test() {
        for (WordBreak2 solution : WordBreak2.values()) {
            for (WordBreak2Sample sample : getSamples(WordBreak2Sample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.wordBreak(sample.input.s, Arrays.asList(sample.input.wordDict)))
                    .containsExactlyElementsIn(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/WordBreak2Test.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class WordBreak2Test : SampledTest() {
    override fun getSampleFilename(): String = "word-break-ii.json"

    @Test
    fun test() {
        WordBreak2.entries.forEach { solution ->
            getSamples<Array<WordBreak2Sample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.wordBreak(input.s, input.wordDict.toList()))
                    .containsAtLeastElementsIn(output)
            }
        }
    }
}
